internal/cmd/commands/connect/ssh.go

```diff
@@ -41,18 +42,49 @@ func (s *sshFlags) defaultExec() string {
 	return strings.ToLower(s.flagSshStyle)
 }
 
-func (s *sshFlags) buildArgs(c *Command, port, ip, addr string) []string {
-	// Might want -t for ssh or -tt but seems fine without it for now...
-	var args []string
-	switch s.flagSshStyle {
+func (s *sshFlags) buildArgs(c *Command, port, ip, addr string) (args, envs []string, retErr error) {
+	var cred usernamePasswordCredentials
+	if c.sessionAuthz != nil {
+		creds, err := parseCredentials(c.sessionAuthz.Credentials)
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error interpreting secret: %w", err)
+		}
+		if len(creds) > 0 {
+			// Just use first credentials returned
+			cred = creds[0]
+		}
+	}
+
+	switch strings.ToLower(s.flagSshStyle) {
 	case "ssh":
+		// Might want -t for ssh or -tt but seems fine without it for now...
 		args = append(args, "-p", port, ip)
 		args = append(args, "-o", fmt.Sprintf("HostKeyAlias=%s", c.sessionAuthzData.HostId))
+
+	case "sshpass":
+		if cred.Password == "" {
+			return nil, nil, errors.New("Credential is required when using sshpass")
+		}
+		// sshpass requires that the password is passed as env-var "SSHPASS"
+		// when the using env-vars.
+		envs = append(envs, fmt.Sprintf("SSHPASS=%s", cred.Password))
+		args = append(args, "-e", "ssh")
+		args = append(args, "-p", port, ip)
+
+		// sshpass cannot handle host key checking, disable localhost key verification
+		// to avoid error: 'SSHPASS detected host authentication prompt. Exiting.'
+		args = append(args, "-o", "NoHostAuthenticationForLocalhost=yes")
```

Comment:
I wonder if we should always turn this on, not just for sshpass. Right now we set the host name of the local connection to the host ID in order to not keep seeing host authentication mismatches, but adding this would help too.